New Resource: Azure Front Door (#3933)
terraform-providers/terraform-provider-azurerm: website/docs/r/front_door.html.markdown

---
layout: "azurerm"
page_title: "Azure Resource Manager: azurerm_front_door"
sidebar_current: "docs-azurerm-resource-front-door"
description: |-

>   Manage Azure FrontDoor instance.

Reviewer: ```suggestion
  Manage an Azure FrontDoor instance.
```

Author: fixed

---

# azurerm_front_door

> Manage Azure FrontDoor instance.

Reviewer: ```suggestion
Manage an Azure FrontDoor instance.
```

Author: Fixed

---
layout: "azurerm"
page_title: "Azure Resource Manager: azurerm_front_door"
sidebar_current: "docs-azurerm-resource-front-door"
description: |-
  Manage an Azure Front Door instance.
---

# azurerm_front_door

> Manage an Azure Front Door instance.

Author: I added a little blurb, but I'm not sure its what you are looking for.

> Manage an Azure Front Door instance.

Reviewer: I more meant what bits are required for what. As we create these load balancing blocks, and then refer to them elsewhere. Its not typical of any other azure resource and i think we should maybe explain it a little? wdyt?

  }

  frontend_endpoint {
    name                              = "exampleFrontendEndpoint1"
    host_name                         = "example-FrontDoor.azurefd.net"
    custom_https_provisioning_enabled = false
  }
}
```

## Argument Reference

The following arguments are supported:

* `name` - (Required) Name of the Front Door which is globally unique. Changing this forces a new resource to be created.

* `resource_group` - (Required) Name of the Resource group within the Azure subscription. Changing this forces a new resource to be created.

* `location` - (Required) Resource location. Changing this forces a new resource to be created.

* `backend_pool` - (Required) A `backend_pool` block as defined below.

* `backend_pool_health_probe` - (Required) A `backend_pool_health_probe` block as defined below.

* `backend_pool_load_balancing` - (Required) A `backend_pool_load_balancing` block as defined below.

* `load_balancer_enabled` - (Optional) Operational status of the Front Door load balancer. Permitted values are `true` or `false` Defaults to `true`.

* `friendly_name` - (Optional) A friendly name for the Front Door service.

> * `frontend_endpoint` - (Required) A `frontend_endpoint` block as defined below.

Author: This is the actual `frontend_endpoint` that the `routing_rule.frontend_endpoint` references. The text in the routing_rule is not correct, I'll fix that and it should make more sense. 😃

* `routing_rule` - (Required) A `routing_rule` block as defined below.

* `tags` - (Optional) Resource tags.

---

The `backend_pool` block supports the following:

> * `id` - The Resource ID of the Azure Front Door Backend Pool.

Author: It's an exported attribute.

* `name` - (Required) The name of the `Backend Pool`.

* `backend` - (Required) A `backend` block as defined below.

* `load_balancing_name` - (Required) The name of the `backend_pool_load_balancing` to use for the `Backend Pool`.

> * `health_probe_name` - (Required) The name of the `backend_pool_health_probe` to use for the `Backend Pool`.

Author: Yep. fixed.

---

The `backend` block supports the following:

> * `id` - The Resource ID of the Azure Front Door Backend.

Author: It's an exported attribute.

* `address` - (Required) Location of the backend (IP address or FQDN)

* `host_header` - (Required) The value to use as the host header sent to the backend.

* `http_port` - (Required) The HTTP TCP port number. Possible values are between `1` - `65535`.

* `https_port` - (Required) The HTTPS TCP port number. Possible values are between `1` - `65535`.

* `priority` - (Optional) Priority to use for load balancing. Higher priorities will not be used for load balancing if any lower priority backend is healthy. Defaults to `1`.

* `weight` - (Optional) Weight of this endpoint for load balancing purposes. Defaults to `50`.

---

The `frontend_endpoint` block supports the following:

> * `id` - The Resource ID of the Azure Front Door Frontend Endpoint.

Author: It's an exported attribute.

* `name` - (Required) The name of the Frontend Endpoint.

* `host_name` - (Required) The host name of the Frontend Endpoint. Must be a domain name.

* `session_affinity_enabled` - (Optional) Whether to allow session affinity on this host. Valid options are `true` or `false` Defaults to `false`.

* `session_affinity_ttl_seconds` - (Optional) The TTL to use in seconds for session affinity, if applicable. Defaults to `0`.

[//]: * "* `web_application_firewall_policy_link_id` - (Optional) The `id` of the `web_application_firewall_policy_link` to use for this Frontend Endpoint."

---

The `backend_pool_health_probe` block supports the following:

> * `id` - The Resource ID of the Azure Front Door Backend Health Probe.

Author: and the same they are all exported attributes.

* `id` - The Resource ID of the Azure Front Door Backend Load Balancer.

* `name` - (Required) The name of the Azure Front Door Backend Load Balancer.

* `sample_size` - (Optional) The number of samples to consider for load balancing decisions. Defaults to `4`.

* `successful_samples_required` - (Optional) The number of samples within the sample period that must succeed. Defaults to `2`.

* `additional_latency_milliseconds` - (Optional) The additional latency in milliseconds for probes to fall into the lowest latency bucket. Defaults to `0`.

---

The `routing_rule` block supports the following:

* `id` - The Resource ID of the Azure Front Door Backend Routing Rule.

* `name` - (Required) The name of the Front Door Backend Routing Rule.

* `frontend_endpoints` - (Required) A `frontend_endpoint` block as defined below.

* `accepted_protocols` - (Optional) Protocol schemes to match for the Backend Routing Rule. Defaults to `Http`.

* `patterns_to_match` - (Optional) The route patterns for the Backend Routing Rule. Defaults to `/*`.

* `enabled` - (Optional) `Enable` or `Disable` use of this Backend Routing Rule. Permitted values are `true` or `false`. Defaults to `true`.

---

> The `frontend_endpoint` block supports the following:

Author: Yep. fixed.

* `enable_custom_https_provisioning` - (Required) Name of the Frontend Endpoint.

---

The `backend_pool_health_probe` block supports the following:

* `name` - (Required) The name of the Azure Front Door Backend Health Probe.

* `path` - (Optional) The path to use for the Backend Health Probe. Default is `/`.

* `protocol` - (Optional) Protocol scheme to use for the Backend Health Probe. Defaults to `Http`.

* `interval_in_seconds` - (Optional) The number of seconds between health probes. Defaults to `120`.

---

The `backend_pool_load_balancing` block supports the following:

* `name` - (Required) The name of the Azure Front Door Backend Load Balancer.

* `sample_size` - (Optional) The number of samples to consider for load balancing decisions. Defaults to `4`.

* `successful_samples_required` - (Optional) The number of samples within the sample period that must succeed. Defaults to `2`.

* `additional_latency_milliseconds` - (Optional) The additional latency in milliseconds for probes to fall into the lowest latency bucket. Defaults to `0`.

---

The `routing_rule` block supports the following:

* `name` - (Required) The name of the Front Door Backend Routing Rule.

> * `frontend_endpoints` - (Required) The name property of the `frontend_endpoint` block whithin this resource to associate with this `routing_rule`.

Reviewer: Could we elaborate that multiple can go here? I think this might be worded better as
```suggestion
* `frontend_endpoints` - (Required) The names of the `frontend_endpoint` blocks whithin this resource to associate with this `routing_rule`.
```

or

```suggestion
* `frontend_endpoints` - (Required) The name properties of the `frontend_endpoint` blocks whithin this resource to associate with this `routing_rule`.
```
